homeassistant/components/intellifire/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from intellifire4py import IntellifirePollData
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_ATTRIBUTION,
+    DEVICE_CLASS_TEMPERATURE,
+    TEMP_CELSIUS,
+    TIME_MINUTES,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import IntellifireDataUpdateCoordinator
+from .const import DOMAIN
+
+
+@dataclass
+class IntellifireSensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[
+        [IntellifirePollData], int
+    ]  # Although sensours could have a variety of different return values - all the ones below are only reutrning ints
+
+
+@dataclass
+class IntellifireSensorEntityDescription(
+    SensorEntityDescription, IntellifireSensorEntityDescriptionMixin
+):
+    """Describes a sensor sensor entity."""
+
+
+ATTRIBUTION = "Data provided by unpublished Intellifire API"
+
+
+INTELLIFIRE_SENSORS: tuple[IntellifireSensorEntityDescription, ...] = (
+    IntellifireSensorEntityDescription(
+        key="flame_height",
+        icon="mdi:fire-circle",
+        name="Flame Height",
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda data: data.flameheight,
+    ),
+    IntellifireSensorEntityDescription(
+        key="temperature",
+        icon="mdi:car-brake-temperature",
+        name="Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=DEVICE_CLASS_TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        value_fn=lambda data: data.temperature_c,
+    ),
+    IntellifireSensorEntityDescription(
+        key="target_temp",
+        icon="mdi:thermometer-lines",
```

Comment:
This entity has a device class, and thus should not have an icon

```suggestion
```

Reply:
I've been struggling with the fact there are multiple temp sensors with the same icons -> I'll prob nix the device class and that way I can use separate icons to differentiate the different temps. Is there a downside to doing this?

Reply:
So, we try to streamline experiences, by displaying, e.g., temperature sensors in the same way. Additionally, end-user can customize if they want.